HW1/P4/P4.py

```diff
@@ -0,0 +1,42 @@
+import csv
+from itertools import permutations
+import findspark
+findspark.init()
+import pyspark
+from P4_bfs import bfs1, bfs2
+from shortest_path_test import shortest_path
+
+sc = pyspark.SparkContext(appName='marvel')
+logger = sc._jvm.org.apache.log4j
+logger.LogManager.getLogger("org"). setLevel( logger.Level.ERROR )
+logger.LogManager.getLogger("akka").setLevel( logger.Level.ERROR )
+
+sc.unvisited = sc.accumulator(0)
+sc.explored = sc.accumulator(0)
+
+def main():
+	adjacencies = load_adjacencies(cache=False)
+
+	characters = ['CAPTAIN AMERICA', 'MISS THING/MARY', 'ORWELL']
+	for c in characters:
+		print 'BFS starting from %s touches %d nodes' % (c, bfs2(sc, adjacencies, c))
+
+
+def load_adjacencies(fname='source.csv', cache=False):
+	with open('source.csv') as rfile:
```

Comment:
Nice pre-processing :)
I'm wondering though, given the nature of this problem set, would it have been at all useful to parallelize this? I guess this is such a fast, simple action that it doesn't really matter, but perhaps loading into a textFile and using a map() instead might have been a better approach, just for the purposes of this assignment